### ISSUE TITLE
Links in Headings within diary pages break links on diary index

### ISSUE THREAD

**Issue by fancsali** [vimwiki/vimwiki#1408]
Prior to submitting a new issue make sure to complete these steps:

- [X] Include the VimWiki settings from your `.vimrc`
```
let g:vimwiki_list = [{'path': '~/vimwiki/', 'syntax': 'markdown', 'ext': '.md', 'auto_diary_index': 1, 'diary_caption_level': 6, 'auto_tags': 1, 'listsyms': ' .X'}]
let g:vimwiki_global_ext = 0
let g:vimwiki_auto_header = 1
let g:vimwiki_emoji_enable = 0
```

- [X] Include the syntax you are using (default / Markdown / MediaWiki) -- Markdown
- [X] Provide a detailed description of the problem including **steps to reproduce the issue**.
  - Enable diary captions/headings to be show on the diary page: set `'diary_caption_level': 6`, for example
  - Go to today's note: `<Leader>w<Leader>w`
  - Create a heading, that is a link (either by hand, or typing something and then hitting `<CR>`): `## [FOO](/bar)`
  NOTE: Due to `g:vimwiki_auto_header = 1` there'll be a top level ("h1") heading above this with the relevant date)
  - Go back to the index: `<Leader>wi`
  - Observe the link to the heading in question to be broken: `- [[FOO](/bar)](2024-06-05#[FOO](/bar))`; or with the concealment outside of the cursor row: `- FOO](2024-06-05#FOO` (also the last part of that gets an underscore)

In other words, the whole Markdown "code" will be interpreted as the "label" and the target of the link gets generated correctly to point to the heading in the diary. As opposed to only the label being taken, and the original link ignore.

- [X] Include the output of `:VimwikiShowVersion`.
```
Version: 2024.01.24
Os: Linux
Vim: 901
Branch: HEAD
Revision: fde35bb
Date: 2024-01-24 20:14:13 -0700
```
